misc/vtk/include/inviwo/vtk/processors/vtkdatasetinformation.h

```diff
@@ -78,15 +84,37 @@ class IVW_MODULE_VTK_API VTKDataSetInformation : public Processor {
     public:
         ArrayInformationProperty() = delete;
         ArrayInformationProperty(const std::string& arrayName, const std::string& identifier,
-                                 const std::string& dataType, const std::string& numberOfComponents)
+                                 const vtkDataArray* array)
             : CompositeProperty(identifier, arrayName)
-            , dataType_(identifier + "dataType", "Data type", dataType)
-            , numberOfComponents_(identifier + "numberOfComponents", "Components",
-                                  numberOfComponents) {
-            addProperty(dataType_);
-            addProperty(numberOfComponents_);
+            , dataType_(identifier + "dataType", "Data type")
+            , numberOfComponents_(identifier + "numberOfComponents", "Components")
+            , componentInformation_("componentInformation", "Component info") {
+
+            dataType_.set(std::string{array->GetDataTypeAsString()});
+            numberOfComponents_.set(std::to_string(array->GetNumberOfComponents()));
+
+            addProperties(dataType_, numberOfComponents_, componentInformation_);
+
             dataType_.setReadOnly(true);
             numberOfComponents_.setReadOnly(true);
+
+            for (auto i{0}; i < array->GetNumberOfComponents(); ++i) {
+                auto compInfo = new CompositeProperty(StrBuffer{"component{}", i}.c_str(),
+                                                      StrBuffer{"Component {}", i}.c_str());
+
+                auto name = new StringProperty(StrBuffer{"name{}", i}.c_str(), "Name",
+                                               array->GetComponentName(i));
+                name->setReadOnly(true);
+                auto numValues =
+                    new StringProperty(StrBuffer{"numVal{}", i}.c_str(), "Number of tuples",
+                                       std::to_string(array->GetNumberOfTuples()));
```

Comment:
The intended use case for StrBuffer is when ones wants to reuse the same buffer to format a number of strings removing the need for lots of string allocations. see for example https://github.com/inviwo/inviwo/blob/eeaa70bd8ed0047769bf8b723a3240d12df00dcf/modules/opengl/src/shader/shaderutils.cpp#L98-L110 

Here you are much better of just using fmt::format directly 
```suggestion
                auto compInfo = new CompositeProperty(fmt::format("component{}", i),
                                                      fmt::format("Component {}", i));

                auto name = new StringProperty(fmt::format("name{}", i), "Name",
                                               array->GetComponentName(i));
                name->setReadOnly(true);
                auto numValues =
                    new StringProperty(fmt::format("numVal{}", i), "Number of tuples",
                                       std::to_string(array->GetNumberOfTuples()));
```